tests/lean/let1.lean.expected.out

```diff
@@ -1,20 +1,20 @@
 let bool : Type := Prop,
     and : bool → bool → Prop := λ (p q : bool), Π (c : bool), (p → q → c) → c,
-    and_intro : Π (p q : bool), p → q → and p q :=
+    and_intro : Π (p q : bool), p → ∀ (_ : q), and p q :=
       λ (p q : bool) (H1 : p) (H2 : q) (c : bool) (H : p → q → c), H H1 H2,
     and_elim_left : Π (p q : bool), and p q → p := λ (p q : bool) (H : and p q), H p (λ (H1 : p) (H2 : q), H1),
     and_elim_right : Π (p q : bool), and p q → q := λ (p q : bool) (H : and p q), H q (λ (H1 : p) (H2 : q), H2)
 in and_intro :
   ∀ (p q : Prop), p → q → ∀ (c : Prop), (p → q → c) → c
-let1.lean:19:17: error: invalid let-expression, term
+let1.lean:21:17: error: invalid let-expression, term
   λ (p q : bool) (H1 : p) (H2 : q) (c : bool) (H : p → q → c), H H1 H2
 has type
   Π (p q : bool), p → q → Π (c : bool), (p → q → c) → c
 but is expected to have type
-  Π (p q : bool), p → q → and q p
+  Π (p q : bool), p → ∀ (_ : q), and q p
```

Comment:
This particular file is odd since it's using this let-bound `bool` synonym for `Prop`. There's a `Note(kmill)` in the corresponding test file, but I'll take another look to see if this is due to any subtle bugs.

Reply:
As far as I can tell, it's just that the `is_prop` test for `and q p` is false but for `∀ (_ : q), and q p` it's true.

Here's an example that suggests this theory is correct:
```lean
section
abbreviation bool' := Sort 0

#check let --bool'                      := Sort 0,
          and' (p q : bool')         := ∀ c : bool', (p → q → c) → c
          in ∀ (p q : bool'), q → and' p q
/-
let and' : bool' → bool' → Prop := λ (p q : bool'), ∀ (c : bool'), (p → q → c) → c
in ∀ (p q : bool'), q → and' p q :
  Prop
-/
end

#check let bool'                      := Sort 0,
          and' (p q : bool')         := ∀ c : bool', (p → q → c) → c
          in ∀ (p q : bool'), q → and' p q
/-
let bool' : Type := Prop,
    and' : bool' → bool' → Prop := λ (p q : bool'), Π (c : bool'), (p → q → c) → c
in ∀ (p q : bool') (_ : q), and' p q :
  Prop
-/
```
In the first case, the relaxed weak head normal form is able to see that `bool'` is `Prop`, but in the second it can't.

Reply:
I've added some simple tests that point out this limitation.

Reply:
Thanks! I don't think this is a big deal, especially since the non-weird two-binder cases work fine.